computedfields/graph.py

```diff
@@ -437,20 +437,42 @@ def resolve_dependencies(self, computed_models):
         for model, fields in computed_models.items():
             local_deps.setdefault(model, {})    # always add to local to get a result for MRO
             for field, real_field in fields.items():
-
-                # FIXME: multi table inheritance needs special relation path translation here
-                # this should also fix the weird o2o rel access below
-                # Example from test cases:
-                # MtBase.pulled, gets inherited to MtDerived.pulled
-                # --> depends contains ['mtderived', [...]] --> not valid on the derived model anymore
-                if real_field.model != model and not model._meta.abstract:
-                    print('special case multi table inheritance', model, field)
-
                 fieldentry = global_deps.setdefault(model, {}).setdefault(field, {})
                 local_deps.setdefault(model, {}).setdefault(field, set())
 
                 depends = real_field._computed['depends']
 
+                # fields contributed from multi table model inheritance need patched depends rules,
+                # so the relation paths match the changed model entrypoint
+                if real_field.model != model and not real_field.model._meta.abstract:
```

Comment:
just out of curiosity, why do you have to add the `and not real_field.model._meta.abstract` condition ? From my observations it seemed like fields inherited from an abstract model have their `model` attribute pointing to the Concrete model, not the Abstract model.
So Abstract model fields shouldn't even pass `if real_field.model != model`
In fact, IIRC the abstract model doesn't even trigger its own `class_prepared` signal

Reply:
Yes, it is just a security measure to somewhat cover an implicit behavior here more explicit:
- fields from decoration stage still contain the real source model (thus an abstract model)
- in `models_with_computedfields` we pull their correct model association by equality check, this works correctly, which is abit surprising, as the same (=equal) field pulled from `model._meta.fields` vs. from decoration stage has a different `.model` attribute

I have not looked up django's code regarding filling in the `field.model` attribute, so this here is just a guard to not grab here a totally unkown abstract model by accident.

> In fact, IIRC the abstract model doesn't even trigger its own class_prepared signal

Yeah the resolver doesn't know anything about abstract models, as we only listen for models from that signal. Thats quite different to the old metaclass approach, here those qorkaround for stracts were needed.

Reply:
OK thanks for the explanation.

> which is abit surprising, as the same (=equal) field pulled from model._meta.fields vs. from decoration stage has a different .model attribute

Makes sense when looking at Django's `Field` `__eq__` operator implementation:
```
    def __eq__(self, other):
        # Needed for @total_ordering
        if isinstance(other, Field):
            return self.creation_counter == other.creation_counter
        return NotImplemented
```